src/strategies/math/index.ts

```diff
@@ -12,36 +24,104 @@ export async function strategy(
-    getProvider(options.strategy.network),
-    addresses,
-    options.strategy.params,
-    snapshot
+  const rawOptions: OptionalOptions = migrateLegacyOptions(options);
+  const strategyOptions: Options = validateOptions(rawOptions);
+
+  // Recursively resolve operands
+  const operandPromises: Promise<
+    Record<string, number>
+  >[] = strategyOptions.operands.map((item) =>
+    resolveOperand(item, addresses, space, snapshot)
+  );
+  const resolvedOperands: Record<string, number>[] = await Promise.all(
+    operandPromises
+  );
+
+  const finalResult: Record<string, number> = resolveOperation(
+    strategyOptions.operation,
+    resolvedOperands
   );
 
-  let scoreConverter: (score: number) => number;
-  switch (options.operation) {
-    case 'square-root': {
-      scoreConverter = (score) => Math.sqrt(score);
-      break;
+  return finalResult;
+}
+
+function resolveOperation(
+  operation: Operation,
+  resolvedOperands: Record<string, number>[]
+): Record<string, number> {
+  switch (operation) {
+    case Operation.SquareRoot: {
+      return Object.fromEntries(
+        Object.entries(
+          resolvedOperands[0]
+        ).map(([address, score]: [string, number]) => [
+          address,
+          Math.sqrt(score)
+        ])
+      );
     }
-    case 'cube-root': {
-      scoreConverter = (score) => Math.cbrt(score);
-      break;
+    case Operation.CubeRoot: {
+      return Object.fromEntries(
+        Object.entries(
+          resolvedOperands[0]
+        ).map(([address, score]: [string, number]) => [
+          address,
+          Math.cbrt(score)
+        ])
+      );
     }
-    default: {
-      throw new Error(`Unknown math operation: ${options.operation}`);
+    case Operation.Min: {
+      return Object.fromEntries(
+        Object.entries(
+          resolvedOperands[0]
+        ).map(([address, score]: [string, number]) => [
+          address,
+          Math.min(score, resolvedOperands[1][address])
+        ])
+      );
+    }
+    case Operation.Max: {
+      return Object.fromEntries(
+        Object.entries(
+          resolvedOperands[0]
+        ).map(([address, score]: [string, number]) => [
+          address,
+          Math.max(score, resolvedOperands[1][address])
+        ])
+      );
     }
   }
+}
 
-  return Object.fromEntries(
-    Object.entries(upstreamResult).map(([address, score]) => [
-      address,
-      scoreConverter(score)
-    ])
-  );
+async function resolveOperand(
+  operand: Operand,
+  addresses: string[],
+  space: any,
+  snapshot: any
+): Promise<Record<string, number>> {
+  switch (operand.type) {
+    case OperandType.Strategy: {
+      const strategyOperand: StrategyOperand = operand as StrategyOperand;
+
+      const upstreamResult: Record<string, number> = await strategies[
+        strategyOperand.strategy.name
+      ].strategy(
+        space,
+        strategyOperand.strategy.network,
+        getProvider(strategyOperand.strategy.network),
```

Comment:
We shouldnt be able to change the network, changing network would make "snapshot" value not correct, we should default network instead.

Reply:
Hi @bonustrack, thanks for the feedback.

Yes this makes sense. If the user wants to use data from multiple chains they should simply nest a `mutlichain` strategy inside.

Reply:
@bonustrack Changed to use upstream `network` and `provider`.